### PR TITLE
[Merged by Bors] - feat: the universal property of a disjoint union of topological spaces

### PR DESCRIPTION
**Pull request by dagurtomas**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/ContinuousFunction/Basic.lean
Comment (eric-wieser) [inline Mathlib/Topology/ContinuousFunction/Basic.lean:349]: Can you add a reference to `Equiv.piCurry` in this docstring?

File: Mathlib/Topology/ContinuousFunction/Basic.lean
Comment (eric-wieser) [inline Mathlib/Topology/ContinuousFunction/Basic.lean:382]:
```suggestion
@[simps]
def piEquiv : (∀ i, C(A, X i)) ≃ C(A, ∀ i, X i) where
```

File: Mathlib/Topology/ContinuousFunction/Basic.lean
Comment (eric-wieser) [inline Mathlib/Topology/ContinuousFunction/Basic.lean:381]: This possibly deserves a reference to `Equiv.curry`; I'll let another reviewer decide.

File: Mathlib/Topology/ContinuousFunction/Basic.lean
Comment (eric-wieser) [inline Mathlib/Topology/ContinuousFunction/Basic.lean:342]: I think this is
```suggestion
/--
To give a continuous map out of a disjoint union, it suffices to give a continuous map out of
each term
-/
def sigma (f : ∀ i, C(X i, A)) : C((Σ i, X i), A) where
  toFun := Sigma.curry fun i x => f i x
```
which is probably worth mentioning in the docstring, if you don't want this implementation.

File: Mathlib/Topology/ContinuousFunction/Basic.lean
Comment (eric-wieser) [inline Mathlib/Topology/ContinuousFunction/Basic.lean:342]:
```suggestion
def sigma (f : ∀ i, C(X i, A)) : C((Σ i, X i), A) where
```
```suggestion
@[simps]
def sigma (f : ∀ i, C(X i, A)) : C((Σ i, X i), A) where
```

File: Mathlib/Topology/ContinuousFunction/Basic.lean
Comment (eric-wieser) [reply Mathlib/Topology/ContinuousFunction/Basic.lean:384]: Nevermind, that's not relevant here

File: Mathlib/Topology/ContinuousFunction/Basic.lean
Comment (eric-wieser) [inline Mathlib/Topology/ContinuousFunction/Basic.lean:340]:
```suggestion
each term. This is `Sigma.uncurry fun i x => f i x`, but `aesop` doesn't manage to prove
continuity with that implementation.
```
I think it's enough to say

```suggestion
each term. This is `Sigma.uncurry` for continuous maps.
```

**Review by eric-wieser — APPROVED** [review]
maintainer merge

Thanks!

**Review by jcommelin — APPROVED** [review]
bors d+

**Pull request closed without merging**
